src/toast.toasts.ts

```diff
@@ -0,0 +1,357 @@
+    // Toast.
+    //
+    /* eslint-disable lit-a11y/click-events-have-key-events, lit-a11y/mouse-events-have-key-events */
+    return html`<div
+      aria-label=${this.#localize.term('notifications')}
+      class="component"
+      data-test="component"
+      role="region"
+      tabindex="-1"
+      ${ref(this.#componentElementRef)}
+    >
+      <div class="toasts">
+        ${repeat(
+          this.toasts,
+          (toast) => toast.id,
+          (toast) => {
+            return html`<div
+              aria-labelledby="prefix label description"
+              class=${classMap({
+                toast: true,
+                error: toast.variant === 'error',
+                informational: toast.variant === 'informational',
+                success: toast.variant === 'success',
+                show: Boolean(toast.privateShow),
+                dismissing: Boolean(toast.privateDismissing),
+                'dismissing-via-button': Boolean(
+                  toast.privateDismissingViaButton,
+                ),
+              })}
+              data-test="toast"
+              role="alert"
+              @mouseover=${this.#onToastMouseOver.bind(this, toast)}
+              @mouseout=${this.#onToastMouseOut.bind(this, toast)}
+              @transitionend=${this.#onToastTransitionEnd.bind(this, toast)}
+            >
+              <span class="prefix" id="prefix">
+                ${this.#localize.term(toast.variant)}
+              </span>
+
+              ${choose(
+                toast.variant,
+                [
+                  ['success', () => icons.success],
+                  ['error', () => icons.error],
+                ],
+                () => icons.warningInformational,
+              )}
+
+              <div class="label" data-test="label" id="label">
+                ${toast.label}
+              </div>
+
+              <glide-core-icon-button
+                class="dismiss-button"
+                data-test="dismiss-button"
+                label=${this.#localize.term('dismiss')}
+                variant="tertiary"
+                @click=${this.#onToastDismissButtonClick.bind(this, toast)}
+              >
+                ${xIcon}
+              </glide-core-icon-button>
+
+              ${when(toast.privateDescription, () => {
+                return html`
+                  <div
+                    class="description"
+                    data-test="description"
+                    id="description"
+                    @click=${this.#onDescriptionClick.bind(this, toast)}
+                  >
+                    ${unsafeHTML(toast.privateDescription)}
+                  </div>
+                `;
+              })}
+            </div>`;
+          },
+        )}
+      </div>
+    </div>`;
+  }
+
+  override showPopover() {
+    this.#componentElementRef.value?.showPopover();
+  }
+
+  static async show(toast: GlideCoreToast): Promise<void> {
```

Comment:
1. Toast calls this static method during its `connectedCallback()` and passes itself as an argument. 
2. Toasts appends itself to `document.body`.
3. Toasts goes through the Toast's light DOM, gets the content of every `Element` and `Text` node, then adds that content to the Toast's `privateDescription` property.
4. Toasts adds the Toast to its array of `toasts`.
5. Toasts unhides itself.
6. Toasts sets a timer removing the Toast after its `duration` has expired.